Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIDeclarativeNetRequestCocoa.mm

```diff
@@ -261,6 +264,141 @@
     completionHandler(filteredMatchedRules, std::nullopt);
 }
 
+_WKWebExtensionDeclarativeNetRequestSQLiteStore *WebExtensionContext::declarativeNetRequestDynamicRulesStore()
+{
+    if (!m_declarativeNetRequestDynamicRulesStore)
+        m_declarativeNetRequestDynamicRulesStore = [[_WKWebExtensionDeclarativeNetRequestSQLiteStore alloc] initWithUniqueIdentifier:uniqueIdentifier() directory:m_storageDirectory usesInMemoryDatabase:!storageIsPersistent()];
```

Comment:
While I don't think this will break anything, it will cause incorrect error strings.

Passing `YES` for `usesInMemoryDatabase` will cause this DB to use `session_rules` as the SQL table name. That won't conflict with `dynamic` since they are separate DBs, I think we should have this code:

```
    _storageType = _useInMemoryDatabase ? @"session" : @"dynamic";
    _tableName = [NSString stringWithFormat:@"%@_rules", _storageType];
```

You will want to pass in `storageType` likely as a new enum, and have `usesInMemoryDatabase` be separate.

Reply:
I've:
- created a new storage type enum
- Always use the same table name (since they are different databases)
- Set the _storageType ivar based on the enum value